diff --git a/src/screenshot-application.c b/src/screenshot-application.c
index 672c9ce..5462183 100644
--- a/src/screenshot-application.c
+++ b/src/screenshot-application.c
@@ -48,6 +48,7 @@ G_DEFINE_TYPE (ScreenshotApplication, screenshot_application, GTK_TYPE_APPLICATI
 
 static void screenshot_save_to_file (ScreenshotApplication *self);
 static void screenshot_show_interactive_dialog (ScreenshotApplication *self);
+static void screenshot_back (ScreenshotApplication *self);
 
 struct _ScreenshotApplicationPriv {
   gchar *icc_profile_base64;
@@ -123,7 +124,8 @@ save_pixbuf_handle_success (ScreenshotApplication *self)
 
   if (screenshot_config->interactive)
     {
-      screenshot_close_interactive_dialog (self);
+      //screenshot_close_interactive_dialog (self);
+      screenshot_back (self);
     }
   else
     {
