diff --git a/src/cheese-flash.c b/src/cheese-flash.c
index f60cd56..e0cfe4e 100644
--- a/src/cheese-flash.c
+++ b/src/cheese-flash.c
@@ -104,7 +104,7 @@ cheese_flash_init (CheeseFlash *self)
   gdk_window_input_shape_combine_region (gtk_widget_get_window (GTK_WIDGET (window)), input_region, 0, 0);
   cairo_region_destroy (input_region);
 
-  g_signal_connect (G_OBJECT (window), "draw", G_CALLBACK (cheese_flash_window_draw_event_cb), NULL);
+//  g_signal_connect (G_OBJECT (window), "draw", G_CALLBACK (cheese_flash_window_draw_event_cb), NULL);
   priv->window = window;
 }
 
